### ISSUE TITLE
Netgear series GS748T makes Oxidized crash? when connected via telnet

### ISSUE THREAD

**Issue by tuxpilot** [ytti/oxidized#2223]
Hi.

Before anything, i'm new to Oxidized.
I love the possibilities it offers, but i can't make it work properly.

Context :
VM, Debian10 (amd64), oxidized : Latest version, installed as a service on the VM.

Target to be saved : 
Netgear GS748Tv5.
SSH not implemented on this switch, telnet only on port 60000

VM and Switch are not on the same Network, but all the ports are open between the 2 devices, firewalls, etc.

I've managed to install Oxidized and make it run, i can go on the webUI.
But everytime i run it, it starts properly to telnet to the switch, input the username and the password, then in the logs i can see the prompt ending by '>' and waiting further instructions,    but.... nothing.
Oxidized had crashed.
If i put a wrong password in the router.db  in the logs i get a message : 
"Applying Interface configuration, please wait ..."

I get the very same message if i telnet the switch manualy and input a wrong password.
If i telnet the switch manually fromthe same VM, with the right password, i get in, i can go in enable mode, and do stuff.

This is my config : 
---
username: admin
model: netgear
resolve_dns: false
interval: 3600
use_syslog: true
debug: true
threads: 30
timeout: 60
retries: 3
prompt: !ruby/regexp /^([\w.@-]+[#>]\s?)$/
rest: <debian10_server_ip>:8081
next_adds_job: false
vars:
  enable: true
groups: {}
models: {}
pid: "/etc/oxidized/.config/oxidized/pid"
crash:
  directory: "/etc/oxidized/.config/oxidized/crashes"
  hostnames: false
stats:
  history_size: 20
input:
  default: telnet
  debug: true
  ssh:
    secure: false
  ftp:
    passive: true
  utf8_encoded: true
output:
  default: file
  file:
    directory: "/etc/oxidized/.config/oxidized/configs"
source:
  default: csv
  csv:
    file: "/etc/oxidized/.config/oxidized/router.db"
    delimiter: !ruby/regexp /:/
    map:
      ip: 0
      model: 1
      username: 4
      password: 3
    vars_map:
      telnet_port: 2
      enable: 5
    gpg: false
model_map:
  juniper: junos
  cisco: ios
  netgear: netgear
  dlink:dlink

This is my router.db :
#ip:model:port:password:username:enable
<switch_ip>:netgear:60000:<my_password>:admin:true

This is the log from the telnet log of the switch in Oxidized: 
User:admin
Password:************
(GS748Tv5) >

This is the error message i get from the syslog:
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/nodes.rb: Loading nodes
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: resolving DNS for ...
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: IPADDR <switch_ip>
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: resolving node key 'model', with passed global value of '' and node value 'netgear'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: setting node key 'model' to value 'netgear' from global
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: returning node key 'model' with value 'netgear'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/node.rb: Loading model "netgear"
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Added secret to the commands list
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Added all to the commands list
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Added show version to the commands list
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Added show bootvar to the commands list
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Added show running-config to the commands list
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: resolving node key 'input', with passed global value of 'telnet' and node value ''
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: returning node key 'input' with value 'telnet'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: resolving node key 'output', with passed global value of 'file' and node value ''
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: returning node key 'output' with value 'file'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: resolving node key 'username', with passed global value of '' and node value 'admin'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: setting node key 'username' to value 'admin' from global
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: returning node key 'username' with value 'admin'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: resolving node key 'password', with passed global value of '' and node value '<my_password>'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: setting node key 'password' to value 'password' from global
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: node.rb: returning node key 'password' with value '<my_password>'
Dec  1 14:44:59 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/nodes.rb: Loaded 1 nodes
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/core.rb: Starting the worker...
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: Puma starting in single mode...
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: * Version 3.11.4 (ruby 2.5.5-p157), codename: Love Song
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: * Min threads: 0, max threads: 16
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: * Environment: development
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: * Listening on tcp://<debian10_server_ip>:8081
Dec  1 14:45:00 DEB10-LDB-TOOLS-01 oxidized[55276]: Use Ctrl-C to stop
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/worker.rb: Added / to the job queue
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/worker.rb: 1 jobs running in parallel
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/job.rb: Starting fetching process for  at 2020-12-01 13:45:01 UTC
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/input/cli.rb: Running post_login commands at
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/input/cli.rb: Running post_login command: nil, block: #<Proc:0x00007f8aa825ab40@/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/model/netgear.rb:19> at
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: Telnet: enable @
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/input/cli.rb: Running post_login command: "terminal length 0", block: nil at
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: Telnet: terminal length 0 @
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Collecting commands' outputs
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Executing show version
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: Telnet: show version @
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Executing show bootvar
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: Telnet: show bootvar @
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/model/model.rb Executing show running-config
Dec  1 14:45:01 DEB10-LDB-TOOLS-01 oxidized[55276]: Telnet: show running-config @
Dec  1 14:45:02 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/worker.rb: 1 jobs running in parallel
Dec  1 14:45:02 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/input/cli.rb Running pre_logout commands at
Dec  1 14:45:02 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/node.rb: Oxidized::Telnet ran for  successfully
Dec  1 14:45:02 DEB10-LDB-TOOLS-01 oxidized[55276]: lib/oxidized/job.rb: Config fetched for  at 2020-12-01 13:45:02 UTC
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: Oxidized crashed, crashfile written in /etc/oxidized/.config/oxidized/crash
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: no implicit conversion of nil into String
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/output/file.rb:23:in `join': no implicit conversion of nil into String (TypeError)
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/output/file.rb:23:in `store'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:65:in `process_success'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:47:in `process'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:16:in `block in work'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:16:in `each'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:16:in `work'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:36:in `run'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:29:in `initialize'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:4:in `new'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:4:in `new'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/cli.rb:13:in `run'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /var/lib/gems/2.5.0/gems/oxidized-0.28.0/bin/oxidized:8:in `<top (required)>'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /usr/local/bin/oxidized:23:in `load'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 oxidized[55276]: #011from /usr/local/bin/oxidized:23:in `<main>'
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 systemd[1]: oxidized.service: Main process exited, code=exited, status=1/FAILURE
Dec  1 14:45:03 DEB10-LDB-TOOLS-01 systemd[1]: oxidized.service: Failed with result 'exit-code'.

This is the message i get when Oxidized crashes : 
--------------------------------------------------
2020-12-01 13:45:03 UTC
no implicit conversion of nil into String [TypeError]
--------------------------------------------------
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/output/file.rb:23:in `join'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/output/file.rb:23:in `store'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:65:in `process_success'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:47:in `process'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:16:in `block in work'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:16:in `each'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/worker.rb:16:in `work'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:36:in `run'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:29:in `initialize'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:4:in `new'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/core.rb:4:in `new'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/lib/oxidized/cli.rb:13:in `run'
/var/lib/gems/2.5.0/gems/oxidized-0.28.0/bin/oxidized:8:in `<top (required)>'
/usr/local/bin/oxidized:23:in `load'
/usr/local/bin/oxidized:23:in `<main>'
--------------------------------------------------

This is the options i get with the show command from this switch : 

(GS748Tv5) >show ?

auto-dos                 Display Auto-DoS parameters.
bootpdhcprelay           Display the value of BOOTP/DHCP relay parameters.
class-map                Display DiffServ class information.
classofservice           Display class of service information.
dhcp                     Display DHCP parameters.
dos-control              Display Denial of Service Configuration.
hosts                    To display the default domain name, a list of name
                         server hosts, the static and the cached list of host
                         names and addresses.
ip                       Display IP information.
ipv6                     Display IPv6 information.
network                  Display configuration for inband connectivity.
port-channel             Display port-channel information.
serial                   Display EIA-232 parameters and serial port inactivity
                         timeout.
service-policy           Display DiffServ service statistical information.
slot                     Display information about slot(s).
spanning-tree            Display spanning tree information.
supported                Display information about supported switch and card
                         types.
switchport               Display Switchport Mode.
telnetcon                Display Telnet configuration information.
vlan                     Display VLAN configuration and configure VLANs.

I'm sorry, but i'm not a pro at this, and i'd love to learn so soon i could participate in making debug and making it evolve.

Thank you and congrats for the work already done :).

**Comment by wcarman1** [comment]
You probably got this working already since it's been a while but I just went through a lot of trouble getting my gs752tp setup in oxidized. Here is what I did: 
1. enabling ssh and disabled telnet which helped but didn't fix the problem. I kept getting to the pre_logout stage and the connection would drop off in oxidized
`config t
ip ssh server
ip ssh password-auth`
verify ssh connectivity then
`config t
no ip telnet server
exit
write mem`
2. change your config file so the netgear device using the ciscosmb.rb to grab the config. (I happened to try this because I saw in a nmap scan that it reported this "23/tcp open  telnet  Cisco SG300-28p switch telnetd")

That corrected everything and configs are able to be pulled now from my Netgear gs752tp


my config looks like this: 

`model_map:
  cisco: ios
  juniper: junos
  fortigate: fortios
  netgear: ciscosmb`
